app/src/androidTest/java/com/github/se/eventradar/hosting/CreateEventTest.kt

```diff
@@ -0,0 +1,265 @@
+import com.github.se.eventradar.viewmodel.CreateEventUiState
+import com.github.se.eventradar.viewmodel.CreateEventViewModel
+import com.kaspersky.components.composesupport.config.withComposeSupport
+import com.kaspersky.kaspresso.kaspresso.Kaspresso
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit4.MockKRule
+import io.mockk.verify
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class CreateEventTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSupport()) {
+  @get:Rule val composeTestRule = createComposeRule()
+  // This rule automatic initializes lateinit properties with @MockK, @RelaxedMockK, etc.
+  @get:Rule val mockkRule = MockKRule(this)
+  // Relaxed mocks methods have a default implementation returning values
+  @RelaxedMockK lateinit var mockNavActions: NavigationActions
+  @RelaxedMockK lateinit var mockCreateEventViewModel: CreateEventViewModel
+
+  val mockUser =
+      User(
+          userId = "1",
+          birthDate = "01/01/2000",
+          email = "test@test.com",
+          firstName = "John",
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          accountStatus = "active",
+          eventsAttendeeList = mutableListOf(),
+          eventsHostList = mutableListOf(),
+          friendsList = mutableListOf("2"),
+          profilePicUrl = "",
+          qrCodeUrl = "",
+          bio = "",
+          username = "john_doe")
+
+  private val mockCreateEventUiState =
+      MutableStateFlow(
+          CreateEventUiState(
+              eventName = "Annual Tech Conference",
+              eventPhotoUri = Uri.parse("https://example.com/photo.jpg"),
+              eventDescription = "A comprehensive tech conference.",
+              eventCategory = "Technology",
+              startDate = "2024-01-15",
+              endDate = "2024-01-18",
+              startTime = "10:00 AM",
+              endTime = "5:00 PM",
+              location = "",
+              ticketName = "General Admission",
+              ticketCapacity = "500",
+              ticketPrice = "299.99",
+              organiserList = listOf(mockUser),
+              listOfLocations = listOf(Location(20.0, 20.0, "Location1")),
+              hostFriendsList = listOf(),
+          ))
+
+  @Before
+  fun testSetup() {
+    every { mockCreateEventViewModel.addEvent() } returns Unit
+    every { mockCreateEventViewModel.uiState } returns mockCreateEventUiState
+    composeTestRule.setContent {
+      CreateEventScreen(viewModel = mockCreateEventViewModel, navigationActions = mockNavActions)
+    }
+  }
+
+  @Test
+  fun screenDisplaysAllElementsCorrectly() = run {
```

Comment:
I have added step in all the tests now, but for screen displays all elements correctly, i think it's unnecessary to add because I have named the test tags descriptively, I agree that it should be added to the other tests though and done so! Let me know what you think

Reply:
all good !